### DISCUSSION TITLE
Sharing user agents seen in the wild

### DISCUSSION THREAD

**Discussion by FrancoisDevemy5** [nielsbasjes/yauaa#552] [Q&A] [answered]
Hello team,

I am a happy user of this library.
I have deployed it on my production app and I am now seeing around one million unique user agent strings per month.

Is there a way to provide feedback on some of those user agents where I have seen room for improvement?

Thanks for the hard work on this great tool,
François

**Comment by nielsbasjes** [comment, marked as answer]
Yes, I (me being the entire 'team') greatly appreciate you pointing out the places where the analysis can be improved.
For me the best way is if you simply submit an issue with the full user agent string (and if possible the available Client Hints) and which of the results are incorrect.

Also I would really like to know what it really is I'm looking at. 
What kind of device is this really.
It would greatly help if you have a source of the reason what we are looking at. Some kind of website with a specification or even a link to a git repo with the code are usually very helpful.

Please keep in mind:
- Sometimes you may think the result is incorrect but the reality is that I'm taking these problems into account: https://yauaa.basjes.nl/expect/manipulations/
- Some systems are lying so much it becomes too messy to make any clean conclusions.
- There is a "copy as a testcase" functionality on https://try.yauaa.basjes.nl/ which makes it very easy for me to use you report as a testcase.

![image](https://user-images.githubusercontent.com/240096/166464323-08e75aa5-0e6a-4813-b07e-4767a0f71352.png)

**Reply by FrancoisDevemy5** [reply]
Thanks.
I will create issues for specific User agents strings I find.
Unfortunately I only have access to the user agent string and not that much more information.
Most of the problems I find are related to non-browser things, such as HTTP libraries for native mobile apps and backend servers.

I do also look to [What is my browser](https://developers.whatismybrowser.com/useragents/parse/) and https://user-agents.net/parser for comparison. I guess you know them already.
